frontend/src/metabase/components/Header.jsx

```diff
@@ -39,104 +82,133 @@ export default class Header extends Component {
-                titleAndDescription = (
-                    <TitleAndDescription
-                        title={`New ${this.props.objectType}`}
-                        description={this.props.item.description}
-                    />
-                );
-            }
-        }
-
-        var attribution;
-        if (this.props.item && this.props.item.creator) {
-            attribution = (
-                <div className="Header-attribution">
-                    Asked by {this.props.item.creator.common_name}
-                </div>
-            );
-        }
-
-        var headerButtons = this.props.headerButtons.map((section, sectionIndex) => {
-            return section && section.length > 0 && (
-                <span key={sectionIndex} className="Header-buttonSection flex align-center">
-                    {section.map((button, buttonIndex) =>
-                        <span key={buttonIndex} className="Header-button">
-                            {button}
-                        </span>
-                    )}
-                </span>
-            );
-        });
+        const {
+            item,
+            isEditing,
+            isEditingInfo,
+            editingTitle,
+            editingSubtitle,
+            defaultActions,
+            editingActions,
+            fullscreenActions,
+            objectType,
+            isFullscreen,
+            editBarButtons
+        } = this.props
 
         return (
             <div>
-                {this.renderEditHeader()}
-                {this.renderHeaderModal()}
-                <div className={"QueryBuilder-section flex align-center " + this.props.headerClassName} ref="header">
+                { isEditing && (
+                    <EditBar
+                        title={editingTitle}
+                        subtitle={editingSubtitle}
+                        // TODO - @kdoh - shouldnt bars have standard buttons?
+                        buttons={editBarButtons}
+                    />
+                )}
+                {
+                    // TODO - @kdoh - this is highly specific to the dashbaord
+                    // header and should be refactored out
+                }
+                <HeaderModal
+                    isOpen={!!this.props.headerModalMessage}
+                    height={this.state.headerHeight}
+                    title={this.props.headerModalMessage}
+                    onDone={this.props.onHeaderModalDone}
+                    onCancel={this.props.onHeaderModalCancel}
+                />
+                <div className="py1 lg-py2 xl-py3 wrapper flex align-center" ref="header">
                     <div className="Entity py3">
-                        {titleAndDescription}
-                        {attribution}
+                        {
+                            isEditingInfo
+                                ? (
+                                    <div className="Header-title flex flex-column flex-full bordered rounded my1">
```

Comment:
this "edit name&description" thing could have its own component

Reply:
My instinct here is to pull the inputs into `TitleAndDescription` and pass the edit status but I suppose it could also just be `EditTitleAndDescription` and be separate.